File: goclient/finances/budget-client.go
package main

import (
	"context"
	"log"
	"time"

	"google.golang.org/grpc"

	protoBudget "finances/proto/budget"
)

const (
	address = "localhost:9090" // Replace with your server address
)

func main() {
	// Set up a connection to the server.
	conn, err := grpc.Dial(address, grpc.WithInsecure(), grpc.WithBlock())
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	defer conn.Close()
	client := protoBudget.NewBudgetServiceClient(conn)

	// Contact the server and print out its response.
	id := int32(1) // Replace with a valid ID
	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
	defer cancel()

	req := &protoBudget.BudgetRequest{Id: id}
	res, err := client.GetBudget(ctx, req)
	if err != nil {
		log.Fatalf("could not get budget: %v", err)
	}
	log.Printf("Budget ID: %d, Amount: %f", res.GetId(), res.GetAmount())
}

File: pythonclient/budget-client.py
import grpc
import proto.budget_pb2
import proto.budget_pb2_grpc

def run():
    # Create a channel to connect to the server
    channel = grpc.insecure_channel('localhost:9090')  # Replace with your server address
    stub = proto.budget_pb2_grpc.BudgetServiceStub(channel)

    # Create a request
    request = proto.budget_pb2.BudgetRequest(id=1)  # Replace with a valid ID

    # Call the GetBudget method
    response = stub.GetBudget(request)
    print(f"Budget ID: {response.id}, Amount: {response.amount}")

if __name__ == '__main__':
    run()